profile/src/main/java/org/apache/karaf/profile/assembly/ConfigInstaller.java

```diff
@@ -93,7 +99,54 @@ public void installConfigs(Feature feature, Downloader downloader, ArtifactInsta
                     }
                 }
             }
+            for (ConfigFile configFile : content.getConfigfile()) {
+                if (pidMatching(FilenameUtils.getBaseName(configFile.getFinalname()))) {
+                    installConfig(downloader, configFile);
+                }
+            }
+        }
+    }
+
+    private void installConfig(Downloader downloader, ConfigFile pConfigFile) throws Exception {
+        String path = pConfigFile.getFinalname();
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+
+        Path configFileTarget = homeDirectory.resolve(substFinalName(path));
+        LOGGER.info("      adding config file: {}", homeDirectory.relativize(configFileTarget));
+
+        String location = removeTrailingSlash(stripUrl(pConfigFile.getLocation().trim()));
+        if (!location.startsWith("mvn:")) {
+            LOGGER.warn("Ignoring non maven artifact " + location);
+            return;
+        }
+
+        downloader.download(location, provider -> {
+            synchronized (provider) {
+                Files.createDirectories(configFileTarget.getParent());
+                Files.copy(provider.getFile().toPath(), configFileTarget, StandardCopyOption.REPLACE_EXISTING);
+            }
+        });
+    }
+
+    private String substFinalName(String finalname) {
```

Comment:
Useless (at least not in the target of this class).

Reply:
See comment / documentation above. It can be absolute or relative and use variables.